### ISSUE TITLE
Cannot compile on MAC with meson

### ISSUE THREAD

**Issue by xoraingroup** [rockdaboot/libpsl#171]
Tried all possible ways to compile, but I always get the following errors!

I have installed all dependencies. Using Mac OS Big Sur version 11.0.1.

```bash
FAILED: src/libpsl.5.dylib
cc  -o src/libpsl.5.dylib src/libpsl.5.dylib.p/lookup_string_in_fixed_set.c.o src/libpsl.5.dylib.p/psl.c.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -shared -install_name @rpath/libpsl.5.dylib -compatibility_version 5 -current_version 5 -Wl,-rpath,/usr/local/Cellar/libidn2/2.3.1/lib /usr/local/Cellar/libidn2/2.3.1/lib/libidn2.dylib -lunistring
Undefined symbols for architecture x86_64:
  "_iconv", referenced from:
      _psl_str_to_utf8lower in psl.c.o
  "_iconv_close", referenced from:
      _psl_str_to_utf8lower in psl.c.o
  "_iconv_open", referenced from:
      _psl_str_to_utf8lower in psl.c.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

I couldn't find any such issue in closed issues either. Any help would be appreciated.

**Comment by xoraingroup** [comment]
works with ./configure but not meson. The reason why I am asking this is because gstreamer is using libpsl as subproject and it uses meson and ninja to compile this project. 

The error is occuring and gstreamer compilation fails for this reason.

**Comment by rockdaboot** [comment]
Im not a meson user, but remember that there is a commit that introduced a check for libiconv. AFAIR it was removed shortly after because it broke meson builds for others.
Unsure how to deal with that situation. I leave meson patches to the meson users.

**Comment by xoraingroup** [comment]
I think ideally it should be possible to check if the current OS is MAC or linux/windows to add the check or remove the check.  But let me check :)

**Comment by rockdaboot** [comment]
Are you using the latest commit from the git master branch ?
At least you should try commit 99e7e6a2af40703f80447d6ffb1a0b172edd7eb5, which is not included in the latest libpsl release (0.21.1).

**Comment by xoraingroup** [comment]
Thank you, yes that helped. I have also filed a bug against gst-build repo, if they can find some alternative to resolve this issue.

Nevertheless thank you very much for your time. Really appreciate that :)
